graph/src/data_source/offchain.rs

```diff
@@ -124,29 +128,40 @@ impl DataSource {
             param: Some(param),
             context,
             creation_block: self.creation_block,
-            is_offchain: true,
             done_at: *self.done_at.lock().unwrap(),
+            causality_region: self.causality_region,
         }
     }
 
     pub fn from_stored_dynamic_data_source(
         template: &DataSourceTemplate,
         stored: StoredDynamicDataSource,
     ) -> Result<Self, Error> {
-        let param = stored.param.context("no param on stored data source")?;
+        let StoredDynamicDataSource {
+            manifest_idx,
+            param,
+            context,
+            creation_block,
+            done_at,
+            causality_region,
+        } = stored;
+
+        let param = param.context("no param on stored data source")?;
```

Comment:
Does this apply here? https://graphprotocol.slack.com/archives/C0385VCCSBC/p1663958185550939

Reply:
I don't think the context methods are to blame for newlines in log messages, we just need to be sure to not print errors, or anything, with the debug format `{:?}`. The annoying thing about anyhow is remembering to use the alternate format specifier `{:#}` to print with causes.